src/networking/server/plugin.rs

```diff
@@ -0,0 +1,100 @@
+use std::{net::SocketAddr, sync::Arc};
+
+use bevy::{prelude::{AppBuilder, Plugin}};
+use quinn::{crypto::rustls::TlsSession, generic::Incoming};
+
+use bevy::prelude::IntoQuerySystem;
+use futures::StreamExt;
+
+use tokio::sync::mpsc::{unbounded_channel, UnboundedSender};
+use tracing::{info};
+
+use crate::networking::{events::ReceiveEvent, events::SendEvent, systems::NetworkConnections, systems::SessionEventListenerState, systems::handle_connection, systems::receive_net_events, systems::send_net_events};
+
+// Add this plugin to start a aerver which sends and receives packets to a large number of network connections
+#[derive(Debug)]
+pub struct Network {
+    pub private_key: quinn::PrivateKey,
+    pub certificate: quinn::CertificateChain,
+    pub addr: SocketAddr,
+}
+
+impl Plugin for Network {
+    fn build(&self, app: &mut AppBuilder) {
+        // Create mpsc endpoints for received network events and store them in a resources
+        let (send, recv) = unbounded_channel();
+        app.add_resource(SessionEventListenerState {
+            event_receiver: recv,
+            stream_senders: Default::default(),
+            send_event_reader: Default::default(),
+        });
+
+        app.add_resource::<NetworkConnections>(Default::default());
```

Comment:
Same as previous comment about changing to `app.init_resource::<NetworkConnections>()`